### ISSUE TITLE
[SDK 38] expo-notifications fails to getDevicePushToken on managed Android standalone build with useNextNotificationsApi enabled

### ISSUE THREAD

**Issue by JoaRiski** [expo/expo#9347]
### Summary of Issue 

**EDIT:  This is probably not a bug with Expo, but a misconfiguration. See issue comments for more details and my suggestions on what could be improved.**

On Expo SDK 38, with `android.useNextNotificationsApi` set to true in `app.config.ts`, without custom Firebase credentials, the `getDevicePushToken` results in the following error:

```
Fetching the token failed: Given String is empty or null
```

This could be due to not having included `google-services.json` in app configuration, in which case the documentation should more explicitly mention that to be required.

### Environment - output of `expo diagnostics` & the platform(s) you're targeting

```
  Expo CLI 3.22.3 environment info:
    System:
      OS: Windows 10 10.0.18362
    Binaries:
      Node: 12.16.3 - C:\Program Files\nodejs\node.EXE
      Yarn: 1.22.4 - ~\AppData\Roaming\npm\yarn.CMD
      npm: 6.14.4 - C:\Program Files\nodejs\npm.CMD
    IDEs:
      Android Studio: Version  4.0.0.0 AI-193.6911.18.40.6514223
```
However the above diagnostics output is not where the build happens, as we use expo build services.

Target platforms: Android & iOS

### Reproducible Demo

```ts
import Constants from "expo-constants";
import * as Notifications from "expo-notifications";
import * as Permissions from "expo-permissions";

Notifications.setNotificationHandler({s
    handleNotification: async () => ({
        shouldShowAlert: true,
        shouldPlaySound: false,
        shouldSetBadge: false
    })
});

async function getNotificationPermission() {
    if ((await Permissions.getAsync(Permissions.NOTIFICATIONS)).status == "granted") {
        return true;
    } else {
        return (await Permissions.askAsync(Permissions.NOTIFICATIONS)).status == "granted";
    }
}

async function getPushNotificationToken() {
    if (Constants.isDevice) {
        if (await getNotificationPermission()) {
            return (await Notifications.getExpoPushTokenAsync()).data;
        } else {
            console.warn("Push notification permission not granted");
        }
    }
}
```
 
### Steps to Reproduce

Build the sample code into a standalone Android application with SDK 38 and `useNextNotificationsApi` set to true, and call `getPushNotificationToken()`

The error seems to be returned from here specifically:
https://github.com/expo/expo/blob/master/packages/expo-notifications/android/src/main/java/expo/modules/notifications/tokens/PushTokenModule.java#L72

### Expected Behavior vs Actual Behavior

Expecting the push notification token to be resolved correctly rather than failing with the error given here.

Alternatively if extra configuration is required, the documentation should be updated accordingly.

**Comment by JoaRiski** [comment]
Adding to the report, the error seems to be returned from here specifically:
https://github.com/expo/expo/blob/master/packages/expo-notifications/android/src/main/java/expo/modules/notifications/tokens/PushTokenModule.java#L72

**Comment by JoaRiski** [comment]
It seems that the [setup of FCM even for managed projects is indeede required](https://docs.expo.io/push-notifications/using-fcm/), the dofucmentation isn't just stating this very clearly, and I believe it's only mentioned on that page (out of multiple pages describing push notification setup, including the push notifications overview page).

So if that is the case (setting up FCM credentials is required), this issue should be read as "clarify push notification setup documentation"

Something worth mentioning is that the [push notifications overview](https://docs.expo.io/push-notifications/overview/) even states
> All the hassle with native device information and communicating with APNS (Apple Push Notification Service) or FCM (Firebase Cloud Messaging) is taken care of behind the scenes, so that you can treat iOS and Android notifications the same, saving you time on the front-end, and back-end!

Which is easy to read as "expo takes care of this for you".

TL;DR: Clarifying the documentation is probably a good idea

**Comment by JoaRiski** [comment]
Seems like the credential requirement is also mentioned on the page https://docs.expo.io/push-notifications/push-notifications-setup/#credentials, but even here at the top of the page it's not listed under the "main things we need" -section.

I will leave this issue open regardless even though there's no bug, since there are some changes that could be done:

- Clarify all documentation around push notification setup and usage, making sure the credential requirement step is mentioned where appropriate, probably especially good in sections that basically claim Expo to handle everything for you
- Improve the error message that is given if Firebase is not configured properly

Feel free to close this issue if these are out of scope. Hopefully this thread will be search-engine visible for anyone in the future hitting the same error message.

**Comment by mczernek** [comment]
What I suppose you're missing is FCM configuration for standalone apps. Please follow [this guide](https://docs.expo.io/push-notifications/using-fcm/).

I'll have this issue closed for now, let me know if that doesn't help.
